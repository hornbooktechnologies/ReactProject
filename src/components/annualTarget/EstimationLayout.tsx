import React from 'react'
import { ChartConstant } from "../../utils/AppConstants";
import { EstmationModel } from "../../models/LinechartModel";
import { useTranslation } from "react-i18next";
import "../../assets/scss/annualTarget/_estimationLayout.scss"
const EstimationLayout = ({linechartData, dataEstList, estimate, selectValue}: EstmationModel) =>{
    const { i18n } = useTranslation();
    return(
        <>
        <div className="uk-child-width-1-2@s uk-child-width-1-3@m uk-child-width-1-3@l uk-grid">
          <div className="mb-20">
            <div className="uk-flex uk-flex-middle pl-20">
              <span className="indicate indicate-circle indicate-bordered  bg-orange-80"></span>
              <div className="ml-10">
                <p className="uk-margin-remove font-12">
                  {i18n.t("ANNUAL_TARGET.PLANNED")}
                </p>
                <strong className="uk-margin-remove font-16">
                  {linechartData?.Prefix !== null
                    ? linechartData?.Prefix
                    : null}
                  {` ${dataEstList && dataEstList[0].toLocaleString("en")} ${
                    linechartData?.Suffix
                  } `}
                </strong>
                <p className="uk-margin-remove font-12">
                  {` ${estimate}`}
                </p>
              </div>
            </div>
          </div>
          <div className="mb-20">
            <div className="uk-flex uk-flex-middle pl-20">
              <span className="indicate indicate-circle indicate-bordered bg-red-80"></span>
              <div className="ml-10">
                <p className="uk-margin-remove font-12">
                  {i18n.t("ANNUAL_TARGET.ACTUAL_CURRENT")}
                </p>
                <strong className="uk-margin-remove font-16">
                  {linechartData?.Prefix !== null
                    ? linechartData?.Prefix
                    : null}
                  {` ${dataEstList && dataEstList[1].toLocaleString("en")} ${
                    linechartData?.Suffix
                  } `}
                </strong>
                <p className="uk-margin-remove font-12">
                  {i18n.t("ANNUAL_TARGET.TO_DATE")}
                </p>
              </div>
            </div>
          </div>
          <div className="mb-20">
            <div className="uk-flex uk-flex-middle pl-20">
              <span className="indicate indicate-circle indicate-bordered bg-blue-80"></span>
              <div className="ml-10">
                <p className="uk-margin-remove font-12">
                
                  {selectValue === ChartConstant.ANNUAL_DATA ||
                  selectValue === ChartConstant.CUMULATIVE_DATA
                    ? i18n.t("ANNUAL_TARGET.FORECAST_REMAINING")
                    :  i18n.t("LABELS.FORECAST")}
                </p>
                <strong className="uk-margin-remove font-16">
                  {linechartData?.Prefix !== null
                    ? linechartData?.Prefix
                    : null}
                  {` ${dataEstList && dataEstList[2].toLocaleString("en")} ${
                    linechartData?.Suffix
                  }`}
                </strong>
                <p className="uk-margin-remove font-12">
                {selectValue === ChartConstant.ANNUAL_DATA ||
                  selectValue === ChartConstant.CUMULATIVE_DATA
                    ?  i18n.t("ANNUAL_TARGET.REMAINING")
                    : ""}
                </p>
              </div>
            </div>
          </div>
        </div>
      </>
    )
}

export default EstimationLayout;