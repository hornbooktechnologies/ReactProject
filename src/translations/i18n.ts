import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
