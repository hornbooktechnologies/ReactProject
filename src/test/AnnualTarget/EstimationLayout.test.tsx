import React from "react";
import renderer from "react-test-renderer";
import EstimationLayout from "../../components/annualTarget/EstimationLayout";
import { I18nextProvider } from "react-i18next";
import i18n from "../../i18nForTests";
describe("Component: EstimationLayout", () => {
  it("EstimationLayout renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <EstimationLayout />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
