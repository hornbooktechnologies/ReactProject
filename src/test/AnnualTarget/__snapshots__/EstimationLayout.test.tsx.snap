// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: EstimationLayout EstimationLayout renders correctly 1`] = `
<div
  className="uk-child-width-1-2@s uk-child-width-1-3@m uk-child-width-1-3@l uk-grid"
>
  <div
    className="mb-20"
  >
    <div
      className="uk-flex uk-flex-middle pl-20"
    >
      <span
        className="indicate indicate-circle indicate-bordered  bg-orange-80"
      />
      <div
        className="ml-10"
      >
        <p
          className="uk-margin-remove font-12"
        >
          ANNUAL_TARGET.PLANNED
        </p>
        <strong
          className="uk-margin-remove font-16"
        >
           undefined undefined 
        </strong>
        <p
          className="uk-margin-remove font-12"
        >
           undefined
        </p>
      </div>
    </div>
  </div>
  <div
    className="mb-20"
  >
    <div
      className="uk-flex uk-flex-middle pl-20"
    >
      <span
        className="indicate indicate-circle indicate-bordered bg-red-80"
      />
      <div
        className="ml-10"
      >
        <p
          className="uk-margin-remove font-12"
        >
          ANNUAL_TARGET.ACTUAL_CURRENT
        </p>
        <strong
          className="uk-margin-remove font-16"
        >
           undefined undefined 
        </strong>
        <p
          className="uk-margin-remove font-12"
        >
          ANNUAL_TARGET.TO_DATE
        </p>
      </div>
    </div>
  </div>
  <div
    className="mb-20"
  >
    <div
      className="uk-flex uk-flex-middle pl-20"
    >
      <span
        className="indicate indicate-circle indicate-bordered bg-blue-80"
      />
      <div
        className="ml-10"
      >
        <p
          className="uk-margin-remove font-12"
        >
          LABELS.FORECAST
        </p>
        <strong
          className="uk-margin-remove font-16"
        >
           undefined undefined
        </strong>
        <p
          className="uk-margin-remove font-12"
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;
